icu4c/source/test/iotest/filetst.c

```diff
@@ -315,17 +329,28 @@ static void TestFileFromICU(UFILE *myFile) {
 }
 
 static void TestFile(void) {
-
-    log_verbose("Testing u_fopen\n");
-    TestFileFromICU(u_fopen(STANDARD_TEST_FILE, "w", STANDARD_TEST_LOCALE, NULL));
+    UChar ufilename[296];
+ 
+    log_verbose("Testing u_fopen with STANDARD_TEST_FILE\n");
+    TestFileFromICU(u_fopen(STANDARD_TEST_FILE, "w", STANDARD_TEST_LOCALE, NULL), "u_fopen STANDARD");
+
+    u_uastrcpy(ufilename, MEDIUMNAME_TEST_FILE);
```

Comment:
I wonder if it would be safer to use `u_uastrncpy` with `UPRV_LENGTHOF(ufilename)` for this, ex:

```c++
u_uastrncpy(ufilename, MEDIUMNAME_TEST_FILE, UPRV_LENGTHOF(ufilename)); 
```
Just in the (rare) case somebody ever adds another test case, or perhaps if for some reason the conversion to the host code page needs more room, so that it doesn't overflow the buffer.

Reply:
Good idea, thanks, I addressed this in a slightly different way that also ensures 0 termination of the converted buf.